modules/nlu/src/backend/module-lifecycle/on-server-started.ts

```diff
@@ -50,22 +50,26 @@ const registerMiddleware = async (bp: typeof sdk, state: NLUState) => {
         return next(undefined, false, true)
       }
 
-      let nluResults: sdk.IO.EventUnderstanding
-      const { engine } = state.nluByBot[event.botId]
-      const extractEngine2 = async () => {
-        // eventually if model not loaded for bot languages ==> train or load
-        nluResults = await engine.predict(event.preview, event.nlu.includedContexts)
-        if (nluResults.errored && nluResults.suggestedLanguage) {
-          const model = await getLatestModel(bp.ghost.forBot(event.botId), nluResults.suggestedLanguage)
-          await engine.loadModel(model)
-          // might throw again, thus usage of bluebird retry
-          nluResults = await engine.predict(event.preview, event.nlu.includedContexts)
+      try {
+        const { engine, defaultLanguage } = state.nluByBot[event.botId]
```

Comment:
as discussed, some cases are not covered here. Please extract this and unit test it

Reply:
:+1: